File: LibrarayManagement/Model/Issue.cs

using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LibrarayManagement.Model
{
    public class Issue
    {
        [Key]
        public int IssueId { get; set; }

        
   

        [ForeignKey("Member")]
        public int MemberId { get; set; }
        public Member? Member { get; set; }

        public DateTime IssueDate { get; set; } = DateTime.Now;
        [DisplayFormat(ConvertEmptyStringToNull = true)]
        public DateTime? ReturnDate { get; set; } = null;
        public bool IsReturned { get; set; }
        public List<Book>? Books { get; set; }
    }
}

File: ApiBindAngular/src/app/models/issue.ts
import { Member } from './member';
import { Book } from './book';

export class Issue {
  issueId: number = 0;

  memberId!: number;        // 🔥 REQUIRED (FK)

  member!: Member | null;          // navigation (GET এ আসবে)

  issueDate: string = new Date().toISOString();
  returnDate?: string | null = null;

  isReturned: boolean = false;

  books: Book[] = [];
}
